app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
+      return self.subnet6.vlanid if self.subnet6
```

Comment:
You're right about previous behavior, I wonder if that was bug or not. The empty string often comes from the UI, when user does not fill in the field and generally considers that as not filled. Therefore I think it should return `nil` for blank strings but I might be wrong.